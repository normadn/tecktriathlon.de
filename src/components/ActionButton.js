import React from "react";

const ActionButton = (props) => {
  const roundedStyles = {
    borderRadius: "28px",
    border: `solid 2px ${props.foregroundColor}`,
    paddingLeft: "55px",
    paddingRight: "55px",
    "@media screen and (max-width: 450px)": {
      width: "92%",
      paddingLeft: "4%",
      paddingRight: "4%"
    }
  };

  const rectStyles = {
    borderRadius: "4px",
    border: "none",
    paddingLeft: "25px",
    paddingRight: "25px"
  };

  let commonStyles = {
    ...props.style,
    minHeight: "48px",
    lineHeight: "48px",
    width: props.width,
    textAlign: "center",
    textTransform: "uppercase",
    backgroundColor: props.backgroundColor,
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    color: props.foregroundColor,
    fontSize: "18px",
    transition: "0.5s ease",
    fontWeight: 700,
    ":hover:not(:disabled)": {
      backgroundColor: props.hoverBackgroundColor || props.foregroundColor,
      color: props.backgroundColor
    }
  };

  switch (props.type) {
    case "rounded":
      commonStyles = { ...roundedStyles, ...commonStyles };
      break;
    case "submit":
      commonStyles = { ...roundedStyles, ...commonStyles };
      break;
    case "rect":
      commonStyles = { ...rectStyles, ...commonStyles };
      break;
  }

  return props.link ? (
    <a
      css={{
        ...commonStyles,
        ...props.style,
        display: "inline-block",
        boxSizing: "content-box"
      }}
      data-cy={props.dataCy}
      disabled={props.disabled}
      onClick={props.onClick}
      href={props.link}
    >
      {props.children}
    </a>
  ) : (
    <button
      css={{
        ...commonStyles,
        ...props.style
      }}
      data-cy={props.dataCy}
      disabled={props.disabled}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
};

export default ActionButton;
