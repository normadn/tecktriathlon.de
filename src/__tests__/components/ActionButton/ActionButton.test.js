import React from "react";
import ActionButton from "../../../components/ActionButton";
import { shallow } from "enzyme";

describe("ActionButton", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <ActionButton
          css={{ padding: "10px", margin: "10px" }}
          text="TeSt1"
          backgroundColor="#000000"
          foregroundColor="#ffffff"
          type="rect"
        />
      )
    ).toMatchSnapshot();
  });
});

describe("ActionButton", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <ActionButton
          css={{ padding: "10px", margin: "10px" }}
          text="tEsT2"
          backgroundColor="#000000"
          foregroundColor="#ffffff"
          type="rounded"
        />
      )
    ).toMatchSnapshot();
  });
});
