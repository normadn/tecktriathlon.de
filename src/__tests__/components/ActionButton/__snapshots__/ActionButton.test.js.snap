// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders correctly 1`] = `
<button
  css={
    Object {
      ":hover:not(:disabled)": Object {
        "backgroundColor": "#ffffff",
        "color": "#000000",
      },
      "alignItems": "center",
      "backgroundColor": "#000000",
      "border": "none",
      "borderRadius": "4px",
      "color": "#ffffff",
      "display": "flex",
      "fontSize": "18px",
      "fontWeight": 700,
      "justifyContent": "center",
      "lineHeight": "48px",
      "minHeight": "48px",
      "paddingLeft": "25px",
      "paddingRight": "25px",
      "textAlign": "center",
      "textTransform": "uppercase",
      "transition": "0.5s ease",
      "width": undefined,
    }
  }
/>
`;

exports[`ActionButton renders correctly 2`] = `
<button
  css={
    Object {
      ":hover:not(:disabled)": Object {
        "backgroundColor": "#ffffff",
        "color": "#000000",
      },
      "@media screen and (max-width: 450px)": Object {
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "width": "92%",
      },
      "alignItems": "center",
      "backgroundColor": "#000000",
      "border": "solid 2px #ffffff",
      "borderRadius": "28px",
      "color": "#ffffff",
      "display": "flex",
      "fontSize": "18px",
      "fontWeight": 700,
      "justifyContent": "center",
      "lineHeight": "48px",
      "minHeight": "48px",
      "paddingLeft": "55px",
      "paddingRight": "55px",
      "textAlign": "center",
      "textTransform": "uppercase",
      "transition": "0.5s ease",
      "width": undefined,
    }
  }
/>
`;
